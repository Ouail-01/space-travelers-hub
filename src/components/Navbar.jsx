import { NavLink } from 'react-router-dom';
import logo from '../planet.png';
import '../style/Navbar.css';

function Navbar() {
  return (
    <div className="nav-container">
      <div className="navbar">
        <div className="logo">
          <img src={logo} alt="logo" />
          <h1 className="title">Space Traveler&apos;s Hub</h1>
        </div>
        <ul className="nav-list">
          <li>
            <NavLink to="/rockets">Rockets</NavLink>
          </li>
          <li>
            <NavLink to="/missions">Mission</NavLink>
          </li>
          <div className="line" />
          <li>
            <NavLink to="/my-profile">My Profile</NavLink>
          </li>
        </ul>
      </div>
      <hr />
    </div>
  );
}

export default Navbar;
