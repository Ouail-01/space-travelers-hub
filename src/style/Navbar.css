.nav-container {
  width: 94%;
  margin: auto;
}

.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 10px;
  padding: 15px 20px;
  flex-wrap: wrap;
}

.logo {
  display: flex;
  align-items: center;
  color: #000;
  text-decoration: none;
}

.logo-image {
  height: 70px;
}

.title {
  font-size: 30px;
  margin-left: 20px;
}

.nav-list {
  display: flex;
  align-items: center;
  list-style: none;
  gap: 20px;
}

.nav-list li a {
  text-decoration: none;
  color: #0a7efe;
}

.nav-list li a.active {
  font-weight: bold;
  text-decoration: underline;
}

.line {
  background: rgb(70, 70, 70);
  border: 1px solid #aaa;
  height: 1.5rem;
  width: 0.15rem;
}
