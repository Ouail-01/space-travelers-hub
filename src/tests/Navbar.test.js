import React from 'react';
import Renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

describe('Nav', () => {
  test('renders correctly', () => {
    const tree = Renderer
      .create(
        <Router>
          <Navbar />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('render Rockets correctly', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const rocketsLink = getByText(/Rockets/i);
    fireEvent.click(rocketsLink);
    expect(window.location.pathname).toBe('/rockets');
  });

  test('render Missions correctly', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const missionsLink = getByText(/Mission/i);
    fireEvent.click(missionsLink);
    expect(window.location.pathname).toBe('/missions');
  });

  test('render Dragons correctly', () => {
    const { getByText } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    const dragonsLink = getByText(/Dragons/i);
    fireEvent.click(dragonsLink);
    expect(window.location.pathname).toBe('/dragons');
  });
});
