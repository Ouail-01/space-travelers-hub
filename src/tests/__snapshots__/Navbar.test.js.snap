// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<div
  className="nav-container"
>
  <div
    className="navbar"
  >
    <a
      aria-current="page"
      className="logo active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo-image"
        src="planet.png"
      />
      <h1
        className="title"
      >
        Space Traveler's Hub
      </h1>
    </a>
    <ul
      className="nav-list"
    >
      <li>
        <a
          className=""
          href="/rockets"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/missions"
          onClick={[Function]}
        >
          Mission
        </a>
      </li>
      <li>
        <a
          className=""
          href="/dragons"
          onClick={[Function]}
        >
          Dragons
        </a>
      </li>
      <div
        className="line"
      />
      <li>
        <a
          className=""
          href="/my-profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
  <hr />
</div>
`;
